src/FSharp.Control.AsyncSeq/AsyncSeq.fs

```diff
@@ -102,6 +102,17 @@ module internal Utils =
           Async.StartWithContinuations(a, ok, err, cnc)
           Async.StartWithContinuations(b, ok, err, cnc)
 
+      /// Creates a computation which produces a tuple consiting of the value produces by the first
+      /// argument computation to complete and a handle to the other computation. The second computation
+      /// to complete is memoized.
+      static member internal chooseBoths (a:Async<'T>) (b:Async<'U>) : Async<Choice<'T * Async<'U>, 'U * Async<'T>>> =
+        Async.chooseBoth (a |> Async.map Choice1Of2) (b |> Async.map Choice2Of2)
```

Comment:
TBH I'm sceptical about the way Async.chooseBoth works in the existing implementation.  It looks likely to create an ever-lengthening chain of tasks as you repeatedly choose between the signal and a move.  The signal is only started once, but each time the "move" wins then a delegating task/async is returned. Doesn't feel right, we need to add stress tests for this.